util/stopper_test.go

```diff
@@ -237,3 +236,52 @@ func TestStopperClosers(t *testing.T) {
 		t.Errorf("expected true & true; got %t & %t", tc1, tc2)
 	}
 }
+
+func TestStopperNumTasks(t *testing.T) {
+	s := NewStopper()
+
+	var tasks []PendingTask
+	for i := 0; i < 3; i++ {
+		tasks = append(tasks, s.StartTask())
+		if numTypes, numTasks := len(s.tasks), s.NumTasks(); numTypes != 1 || numTasks != i+1 {
+			t.Errorf("stopper should have %d running tasks, got %d / %+v", i+1, s.NumTasks(), s.tasks)
+		}
+		m := s.RunningTasks()
+		if len(m) != 1 {
+			t.Fatalf("expected exactly one task map entry: %+v", m)
```

Comment:
it's all fairly incremental and if it fails, I usually don't want to sift through the log finding out where it all began.